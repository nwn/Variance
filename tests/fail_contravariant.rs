mod common;

use common::*;

fn contravariant_fail<'a>() {
    let _contra: Contra<Lifetime<'a>> = Contra(
        Contravariant::<Lifetime<'static>>::default(),
    );
}

fn main() {}
