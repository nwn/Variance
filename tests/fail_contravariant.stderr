error[E0495]: cannot infer an appropriate lifetime due to conflicting requirements
 --> $DIR/fail_contravariant.rs:7:9
  |
7 |         Contravariant::<Lifetime<'static>>::default(),
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
note: first, the lifetime cannot outlive the lifetime `'a` as defined on the function body at 5:23...
 --> $DIR/fail_contravariant.rs:5:23
  |
5 | fn contravariant_fail<'a>() {
  |                       ^^
note: ...so that the expression is assignable
 --> $DIR/fail_contravariant.rs:6:41
  |
6 |       let _contra: Contra<Lifetime<'a>> = Contra(
  |  _________________________________________^
7 | |         Contravariant::<Lifetime<'static>>::default(),
8 | |     );
  | |_____^
  = note: expected `common::Contra<common::Lifetime<'a>>`
             found `common::Contra<common::Lifetime<'_>>`
  = note: but, the lifetime must be valid for the static lifetime...
note: ...so that the expression is assignable
 --> $DIR/fail_contravariant.rs:7:9
  |
7 |         Contravariant::<Lifetime<'static>>::default(),
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = note: expected `type_variance::Contravariant<common::Lifetime<'_>>`
             found `type_variance::Contravariant<common::Lifetime<'static>>`
